commcare_export/commcare_hq_client.py

```diff
@@ -83,7 +83,19 @@ def authenticated(self, username=None, password=None, mode=AUTH_MODE_SESSION):
         else:
             raise Exception('Unknown auth mode: %s' % mode)
 
-        return CommCareHqClient(url=self.url, project=self.project, version=self.version, session=session, auth=auth)
+        return self._clone_with(session=session, auth=auth)
+
+    def _clone_with(self, **kwargs):
+        kwargs.setdefault('version', self.version)
+        kwargs.setdefault('session', self.__session)
+        kwargs.setdefault('auth', self.__auth)
+        cloned = CommCareHqClient(
+            self.url,
+            self.project,
+            **kwargs
+        )
+        cloned.set_checkpoint_manager(self._checkpoint_manager, **self._checkpoint_kwargs)
+        return cloned
```

Comment:
I'm new to this codebase, can you walk me through what's going on?  How was it broken before?  What does "clone" mean in this context - get a new client object that's a copy of the last one?

Reply:
^I could use this too!

Reply:
Sure. I'm not sure this is really necessary and may just be a holdover from the original design. As the docstring for this method says: 

    Returns a freshly authenticated CommCareHqClient with a new session.
    This is safe to call many times and each of the resulting clients
    remain independent, so you can log in with zero, one, or many users.

I don't think that's a real requirement since we only ever auth with a single user per process. There are one or two other uses of this class [outside the library](https://github.com/search?q=CommCareHqClient&type=Code) but probably nothing to bear in mind in terms of backwards compatabilty.

However since this is going to be released as a bug fix on an existing release I don't want to make a larger change.